src/main/java/org/opentripplanner/transit/api/request/TripOnServiceDateRequest.java

```diff
@@ -0,0 +1,76 @@
+package org.opentripplanner.transit.api.request;
+
+import java.time.LocalDate;
+import java.util.List;
+import org.opentripplanner.transit.model.framework.FeedScopedId;
+import org.opentripplanner.transit.model.timetable.TripAlteration;
+
+/*
+ * A request for trips on a specific service date.
+ *
+ * This request is used to retrieve TripsOnServiceDates that match the provided criteria.
+ * At least one operatingDay must be provided.
+ */
+public class TripOnServiceDateRequest {
+
+  private final List<LocalDate> operatingDays;
+  private final List<FeedScopedId> authorities;
+  private final List<FeedScopedId> lines;
+  private final List<FeedScopedId> serviceJourneys;
+  private final List<FeedScopedId> replacementFor;
+  private final List<String> privateCodes;
+  private final List<TripAlteration> alterations;
+
+  protected TripOnServiceDateRequest(
+    List<LocalDate> operatingDays,
+    List<FeedScopedId> authorities,
+    List<FeedScopedId> lines,
+    List<FeedScopedId> serviceJourneys,
+    List<FeedScopedId> replacementFor,
+    List<String> privateCodes,
+    List<TripAlteration> alterations
+  ) {
+    if (operatingDays == null || operatingDays.isEmpty()) {
+      throw new IllegalArgumentException("operatingDays must have at least one date");
+    }
+    this.operatingDays = List.copyOf(operatingDays);
+    this.authorities = List.copyOf(authorities);
+    this.lines = List.copyOf(lines);
+    this.serviceJourneys = List.copyOf(serviceJourneys);
+    this.replacementFor = List.copyOf(replacementFor);
+    this.privateCodes = List.copyOf(privateCodes);
+    this.alterations = List.copyOf(alterations);
```

Comment:
I think this code can be made a bit safer. Add the following method to `ListUtils`:
```
/**
   * This method will convert the given collection to an instance of a List. If the
   * input is {@code null} an empty collection is returned. If not the
   * {@link List#copyOf(Collection)}
   * is called.
   */
  public static <T> List<T> nullSafeImmutableList(Collection<T> c) {
    return (c == null) ? List.of() : List.copyOf(c);
  }
```
and use it here:
```suggestion
    this.operatingDays = List.copyOf(operatingDays);
    this.authorities = ListUtils.nullSafeImmutableList(authorities);
    this.lines = ListUtils.nullSafeImmutableList(lines);
    this.serviceJourneys = ListUtils.nullSafeImmutableList(serviceJourneys);
    this.replacementFor = ListUtils.nullSafeImmutableList(replacementFor);
    this.privateCodes = ListUtils.nullSafeImmutableList(privateCodes);
    this.alterations = ListUtils.nullSafeImmutableList(alterations);
```

Reply:
Implementing.